macOS: deprecation errors for NSFileHandlingPanel{Cancel,OK}Button

**OS:** macOS
**Version:** 0.1.2
**Commit/Build:** 1089bd2d56ccd01c7d3a1adc57b4a1488b0fb571

Compiling on macOS yields deprecation errors for the NSFileHandlingPanelCancelButton and NSFileHandlingPanelOKButton:

```
[100%] Building CXX object CMakeFiles/openrct2.dir/src/openrct2-ui/UiContext.macOS.mm.o
/Users/aaron/Projects/OpenRCT2/OpenRCT2/src/openrct2-ui/UiContext.macOS.mm:105:41: error: 'NSFileHandlingPanelCancelButton' is
      deprecated: first deprecated in macOS 10.13 [-Werror,-Wdeprecated-declarations]
                if ([panel runModal] == NSFileHandlingPanelCancelButton){
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                        NSModalResponseCancel
/System/Library/Frameworks/AppKit.framework/Headers/NSSavePanel.h:20:5: note: 'NSFileHandlingPanelCancelButton' has been
      explicitly marked deprecated here
    NSFileHandlingPanelCancelButton NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSModalResponseCancel", 10_0, 10_13) = NSModalRe...
    ^
/Users/aaron/Projects/OpenRCT2/OpenRCT2/src/openrct2-ui/UiContext.macOS.mm:121:41: error: 'NSFileHandlingPanelOKButton' is
      deprecated: first deprecated in macOS 10.13 [-Werror,-Wdeprecated-declarations]
                if ([panel runModal] == NSFileHandlingPanelOKButton)
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                                        NSModalResponseOK
/System/Library/Frameworks/AppKit.framework/Headers/NSSavePanel.h:21:5: note: 'NSFileHandlingPanelOKButton' has been explicitly
      marked deprecated here
    NSFileHandlingPanelOKButton NS_DEPRECATED_WITH_REPLACEMENT_MAC("NSModalResponseOK", 10_0, 10_13) = NSModalResponseOK,
    ^
2 errors generated.
make[2]: *** [CMakeFiles/openrct2.dir/src/openrct2-ui/UiContext.macOS.mm.o] Error 1
make[1]: *** [CMakeFiles/openrct2.dir/all] Error 2
make: *** [all] Error 2
```

--- a/distribution/changelog.txt
+++ b/distribution/changelog.txt
@@ -35,6 +35,7 @@
 - Improved: [#6186] Transparent menu items now draw properly in OpenGL mode.
 - Improved: [#6218] Speed up game start up time by saving scenario index to file.
 - Improved: Load/save window now refreshes list if native file dialog is closed/cancelled.
+- Technical: [#6384] On macOS, address NSFileHandlingPanel deprecation by using NSModalResponse instead.
 
 0.1.1 (2017-08-09)
 ------------------------------------------------------------------------

--- a/src/openrct2-ui/UiContext.macOS.mm
+++ b/src/openrct2-ui/UiContext.macOS.mm
@@ -102,7 +102,8 @@ namespace OpenRCT2 { namespace Ui
                 panel.title = [NSString stringWithUTF8String:desc.Title.c_str()];
                 panel.allowedFileTypes = extensions;
                 panel.directoryURL = [NSURL fileURLWithPath:directory];
-                if ([panel runModal] == NSFileHandlingPanelCancelButton){
+                if ([panel runModal] == NSModalResponseCancel)
+                {
                     return std::string();
                 } else {
                     return panel.URL.path.UTF8String;
@@ -118,7 +119,7 @@ namespace OpenRCT2 { namespace Ui
                 panel.canChooseFiles = false;
                 panel.canChooseDirectories = true;
                 panel.allowsMultipleSelection = false;
-                if ([panel runModal] == NSFileHandlingPanelOKButton)
+                if ([panel runModal] == NSModalResponseOK)
                 {
                     NSString *selectedPath = panel.URL.path;
                     const char *path = selectedPath.UTF8String;

